src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/LambdaRewriter.Analysis.cs

```diff
@@ -374,6 +289,62 @@ void RecordClosureScope(Scope innermost, Scope outermost, Closure closure)
             }
 
             /// <summary>
+            /// Walk up the scope tree looking for a closure.
+            /// </summary>
+            /// <returns>
+            /// A tuple of the found <see cref="Closure"/> and the <see cref="Scope"/> it was found in.
+            /// </returns>
+            public static (Closure, Scope) FindClosureInScope(Scope startingScope, MethodSymbol closureSymbol)
```

Comment:
Hmm also consider simply `GetClosure`. The `InScope` part to me can read as "find a closure inside this scope" which is contrary to the implementation.